File: backend/api/campaigns.py
from fastapi import APIRouter
from datetime import datetime
from storage.campaign_store import list_campaigns

router = APIRouter(prefix="/campaigns", tags=["campaigns"])


@router.get("/")
def get_campaigns():
    """
    Campaigns V2 API (frontend-safe).

    Guarantees:
    - Always returns { campaigns: [] }
    - Normalized enum casing
    - Unix timestamps in seconds
    - Empty-safe
    """

    raw_campaigns = list_campaigns() or []

    campaigns = []

    for c in raw_campaigns:
        campaigns.append(
            {
                "campaign_id": c.get("campaign_id"),
                "entity": c.get("entity"),
                "decision": (
                    c.get("decision").upper()
                    if c.get("decision")
                    else "UNKNOWN"
                ),
                "risk_score": c.get("risk_score", 0),
                "signals": c.get("signals", []),
                "start_ts": (
                    int(c["start_ts"] / 1000)
                    if c.get("start_ts")
                    else None
                ),
                "last_seen_ts": (
                    int(c["last_seen_ts"] / 1000)
                    if c.get("last_seen_ts")
                    else None
                ),
            }
        )

    return {
        "campaigns": campaigns
    }

File: frontend/src/services/campaigns.ts
import { apiGet } from "./api";

/* =========================
   Campaigns V2 Contract
========================= */

export type Campaign = {
  campaign_id: string;
  entity: string;
  decision: "ALLOW" | "CHALLENGE" | "BLOCK" | "UNKNOWN";
  risk_score: number;
  signals: string[];
  start_ts: number | null;     // unix seconds
  last_seen_ts: number | null; // unix seconds
};

export type CampaignsResponse = {
  campaigns: Campaign[];
};

/* =========================
   API
========================= */

export function getCampaignsV2() {
  // @router.get("/")
  return apiGet<CampaignsResponse>("/campaigns/");
}
